Library/Homebrew/command.rb

```diff
@@ -0,0 +1,134 @@
+module Homebrew
+  class Command
+    attr_reader :command_name, :valid_options, :description, :help_output, :man_output
+
+    def self.initialize
+      @valid_options = []
+      @description = nil
+      # TODO: set the @command_name dynamically
+      @command_name = "commands"
```

Comment:
You could set this via another method in the block taken by `options`, e.g.,

```ruby
options do
  command "commands" # or cmd, cmdname, etc
  # ...
end
```

This would be a little bit repetitive since we're already using the `FooCommand` format for command classes, but it could also give us flexibility in aliasing commands